mobile/src/views/HomeView.jsx

```diff
@@ -1,21 +1,34 @@
 import React from 'react';
+import axios from 'axios';
 import {StyleSheet, View, Text, Image, ImageBackground} from 'react-native';
 import DiscoverBar from '../components/home/DiscoverBar';
 import {globalStyleSheet, colors} from '../utilites/Theme';
 import VendorList from '../components/shared/VendorList';
-import {useState} from 'react';
-
-import vendorData from '../data/vendorMockData';
+import {useState, useEffect} from 'react';
+import {options, url} from '../data/vendorMockData';
 import { shuffle } from '../utilites/Shuffle';
+import { getData } from '../data/vendorMockData';
 
 
 
 const HomeView = () => {
 
-  const vendorDataRandom = shuffle(vendorData);
   const [searchPhrase, setSearchPhrase] = useState('');
   const [clicked, setClicked] = useState(false);
-  
+  const [vendorData, setList] = useState([]);
+
+  React.useEffect(() => {
+    const fetchData = async () => {
+        await axios.get(url, options)
+        .then((res) => {
+          setList(res.data)
+        })
+        .catch((error) => console.log(error));
+    }
+    fetchData();
```

Comment:
I think best practices are to not mix .then syntax with await syntax [0], we should pick one and stick to it . I think here .then is the easiest for what you want, since if you made it all await, you would need to wrap it in a try catch as well.

Can do something like this in your useEffect
```
axios.get(url, options)
        .then((res) => {
          setList(res.data)
        })
        .catch((error) => console.log(error))
```

[0] https://maximorlov.com/why-you-shouldnt-mix-promise-then-with-async-await/